### PR TITLE
Python code: Rename format (more)

### PR DESCRIPTION
**Pull request by bje-**
## What does this PR do?

Rename `format` to `frmt` in more cases.

### CODE REVIEW

**Comment by rouault** [comment]
Python 3 failures:
```
  TEST: test_gdalwarp_lib_override_default_output_nodata ... fail (blowup)
Traceback (most recent call last):
  File "pymod/gdaltest_python3.py", line 46, in run_func
    result = func()
  File "./test_gdalwarp_lib.py", line 1794, in test_gdalwarp_lib_override_default_output_nodata
    format='netCDF', creationOptions=['FORMAT=' + format])
TypeError: Can't convert 'builtin_function_or_method' object to str implicitly
```

```
Traceback (most recent call last):
  File "pymod/gdaltest_python3.py", line 46, in run_func
    result = func()
  File "./test_gdal_calc.py", line 261, in test_gdal_calc_py_5
    gdal_calc.Calc('A', A='tmp/test_gdal_calc_py.tif', overwrite=True, quiet=True, outfile='tmp/test_gdal_calc_py_5_1.tif')
  File "/home/travis/build/OSGeo/gdal/autotest/pyscripts/../../gdal/swig/python/scripts/gdal_calc.py", line 399, in Calc
    doit(opts, None)
  File "/home/travis/build/OSGeo/gdal/autotest/pyscripts/../../gdal/swig/python/scripts/gdal_calc.py", line 130, in doit
    if opts.format is None:
AttributeError: 'Values' object has no attribute 'format'
```

**Comment by rouault** [comment]
There are still issues with gdal_calc. You likely need to change https://github.com/OSGeo/gdal/blob/master/gdal/swig/python/scripts/gdal_calc.py#L426 to have  dest="frmt"

```
Traceback (most recent call last):
  File "pymod/gdaltest_python2.py", line 43, in run_func
    result = func()
  File "./test_gdal_calc.py", line 69, in test_gdal_calc_py_1
    test_py_scripts.run_py_script(script_path, 'gdal_calc', '-A tmp/test_gdal_calc_py.tif --calc=A --overwrite --outfile tmp/test_gdal_calc_py_1_1.tif')
  File "../pymod/test_py_scripts.py", line 85, in run_py_script
    return run_py_script_as_py_module(script_path, script_name, concatenated_argv)
  File "../pymod/test_py_scripts.py", line 148, in run_py_script_as_py_module
    exec(script_name + '.main()')
  File "<string>", line 1, in <module>
  File "/home/travis/build/OSGeo/gdal/autotest/pyscripts/../../gdal/swig/python/scripts/gdal_calc.py", line 452, in main
    doit(opts, args)
  File "/home/travis/build/OSGeo/gdal/autotest/pyscripts/../../gdal/swig/python/scripts/gdal_calc.py", line 130, in doit
    if opts.frmt is None:
AttributeError: Values instance has no attribute 'frmt'
```

**Comment by rouault** [comment]
A rebase will also be needed to solve conflicts

**Comment by rouault** [comment]
@bje- Not sure if you saw my above comments

**Comment by bje-** [comment]
@rouault Yes, just have this on the backburner. :smiley:

**Comment by bje-** [comment]
Should be OK now.

**Comment by rouault** [comment]
```
  TEST: pds4_9 ... fail (blowup)
Traceback (most recent call last):
  File "pymod/gdaltest_python2.py", line 43, in run_func
    result = func()
  File "./pds4.py", line 285, in pds4_9
    gdal.Translate(filename, 'data/byte_pds4.xml', frmt='PDS4')
  File "/usr/lib/python2.7/dist-packages/osgeo/gdal.py", line 390, in Translate
    (opts, callback, callback_data) = TranslateOptions(**kwargs)
TypeError: TranslateOptions() got an unexpected keyword argument 'frmt'
```

File: gdal/swig/python/scripts/gdal_calc.py [hunk 1]
```diff
@@ -362,7 +362,7 @@ def doit(opts, args):
 ################################################################
 
 
-def Calc(calc, outfile, NoDataValue=None, type=None, format=None, creation_options=None, allBands='', overwrite=False, debug=False, quiet=False, **input_files):
+def Calc(calc, outfile, NoDataValue=None, type=None, frmt=None, creation_options=None, allBands='', overwrite=False, debug=False, quiet=False, **input_files):
```
Comment (rouault) [inline gdal/swig/python/scripts/gdal_calc.py]: This is part of the GDAL Python API. We should not change this

File: autotest/gdrivers/pds4.py [hunk 2]
```diff
@@ -282,7 +282,7 @@ def pds4_9():
     filename = '/vsimem/out.xml'
     # Test copy of all specialConstants
     with hide_substitution_warnings_error_handler():
-        gdal.Translate(filename, 'data/byte_pds4.xml', format='PDS4')
+        gdal.Translate(filename, 'data/byte_pds4.xml', frmt='PDS4')
```
Comment (rouault) [inline autotest/gdrivers/pds4.py]: gdal.Translate() is part of the GDAL Python API. We should not change this

**Pull request merged**